facilitator/src/aggregation.rs

```diff
@@ -109,7 +107,20 @@ impl<'a> BatchAggregator<'a> {
         let mut invalid_uuids = Vec::new();
         let mut included_batch_uuids = Vec::new();
 
-        let ingestion_header = self.ingestion_header(&batch_ids[0].0, &batch_ids[0].1)?;
+        let mut ingestion_batch_reader: BatchReader<'_, IngestionHeader, IngestionDataSharePacket> =
```

Comment:
I agree we should avoid needing to spell out specializations everywhere -- but rather than what you suggest, I instead add the header/packet types to `Batch`, and let the compiler figure out the specific type based on which `Batch::new_foo` is called. This works since we always know the "type" of batches that we handle statically. This way, we don't have to write out the types at all in most common cases. (aspirationally, I would eventually like to remove `Batch{Reader,Writer}` entirely, opting for `Batch::read` and `Batch::write` instead, but that's a few more steps down the line & I may never get that far along.)

When embedded as a field in a struct, we do have to spell out the specific type. That's uncommon enough that I'm OK with it.

Reply:
One other upside: we can no longer create a `Batch{Reader,Writer}` reading messages that do not match the type expected by the `Batch`. The downside of this is that I had to refactor the tests in `batch.rs` which were doing just that as a time-saving measure. 😆